### PR TITLE
Feature/frontier persistence hook in

### PR DESCRIPTION
**Pull request by johnwu93**
Thank you for contributing to Coda! Please see `CONTRIBUTING.md` if you haven't
yet.

Explain your changes here.

Explain how you tested your changes here.

Checklist:

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

File: src/app/cli/src/coda_worker.ml [hunk 1]
```diff
@@ -277,6 +277,10 @@ module T = struct
           let module Run = Run (Config) (Main) in
           let%bind trust_dir = Unix.mkdtemp (conf_dir ^/ "trust") in
           let receipt_chain_dir_name = conf_dir ^/ "receipt_chain" in
+          let%bind transition_frontier_location =
+            Unix.mkdtemp (conf_dir ^/ "transition_frontier")
+          in
+          let%bind trust_dir = Unix.mkdtemp (conf_dir ^/ "trust") in
```
Comment (cmr) [inline src/app/cli/src/coda_worker.ml]: this line got duplicated

File: src/lib/transition_frontier_persistence/intf.ml [hunk 2]
```diff
@@ -76,10 +83,13 @@ module type Worker = sig
   val deserialize :
        t
     -> root_snarked_ledger:root_snarked_ledger
-    -> consensus_local_state:Consensus.Local_state.t
+    -> consensus_local_state:consensus_local_state
```
Comment (cmr) [inline src/lib/transition_frontier_persistence/intf.ml]: why this change? consensus isn't a functor, we shouldn't be functoring over bits of it afaik

File: src/app/cli/src/transition_frontier_persistence_rpc_worker.ml [hunk 3]
```diff
@@ -0,0 +1,109 @@
+open Core
+open Async
+open Coda_base
```
Comment (cmr) [inline src/app/cli/src/transition_frontier_persistence_rpc_worker.ml]: Transition_frontier_persistence already has a dependence on `coda_base`. I do think this functor is unnecessary and `Transition_frontier_persistence` should define its own RPC worker.

File: src/app/cli/src/coda_main.ml [hunk 4]
```diff
@@ -612,7 +606,23 @@ struct
...
 
     let max_length = max_length
-  end)
+  end
+
+  module Transition_frontier =
+    Transition_frontier.Make (Transition_frontier_inputs)
+
+  module Transition_frontier_persistence = struct
+    module Worker = Transition_frontier_persistence_rpc_worker.Make (struct
```
Comment (cmr) [inline src/app/cli/src/coda_main.ml]: This seems to have the same inputs as `Transition_frontier_persistence`, are you sure this functor is necessary?

File: src/lib/transition_frontier_persistence/intf.ml [hunk 5]
```diff
@@ -104,6 +114,36 @@ module type Main_inputs = sig
...
+
+  type diff_hash
+
+  val listen_to_frontier_broadcast_pipe :
+       logger:Logger.t
+    -> frontier option Broadcast_pipe.Reader.t
+    -> worker
+    -> unit Deferred.t
+
+  (* TODO: Lol this cannot be polymorphic. Don't actually get any wins if this was polymorphic *)
```
Comment (cmr) [inline src/lib/transition_frontier_persistence/intf.ml]: are you talking about the `'output` type? is this TODO done?

File: src/lib/transition_frontier_persistence/intf.ml (hunk 2 above)
Comment (johnwu93) [reply src/lib/transition_frontier_persistence/intf.ml]: oh, I had some complications hooking in the persistence module into coda_main. Namely, in Coda_main, all the modules were interacting with protocol/coda_pow's `Consesus_mechanism`. As a result, I could not use the already instantiated `Consensus` module.

File: src/lib/transition_frontier_persistence/intf.ml (hunk 5 above)
Comment (johnwu93) [reply src/lib/transition_frontier_persistence/intf.ml]: yep, I deleted it and is unnecessary.
